tests/aws/services/cloudwatch/test_cloudwatch.py

```diff
@@ -988,6 +988,221 @@ def test_set_alarm(self, sns_create_topic, sqs_create_queue, aws_client, cleanup
+            sleep_before=1,
+            sqs_client=aws_client.sqs,
+            queue_url=queue_url_ok,
+            expected_topic_arn=topic_arn_ok,
+            alarm_name=composite_alarm_name,
+            alarm_description=composite_alarm_description,
+            expected_state="OK",
+            expected_triggering_child_arn=alarm_1_arn,
+            expected_triggering_child_state="OK",
+        )
+
+        composite_alarms_list = aws_client.cloudwatch.describe_alarms(
+            AlarmNames=[composite_alarm_name], AlarmTypes=["CompositeAlarm"]
+        )
+        composite_alarm_in_ok_when_alarm_1_back_to_ok = composite_alarms_list["CompositeAlarms"][0]
+        snapshot.match(
+            "composite-alarm-in-ok-when-alarm-1-is-back-to-ok",
+            composite_alarm_in_ok_when_alarm_1_back_to_ok,
+        )
+
+        # trigger alarm 2 - composite one should go again into ALARM state
+        aws_client.cloudwatch.set_alarm_state(
+            AlarmName=alarm_2_name, StateValue="ALARM", StateReason="trigger alarm 2"
+        )
+
+        retry(
+            check_composite_alarm_message,
+            retries=PUBLICATION_RETRIES,
+            sleep_before=1,
+            sqs_client=aws_client.sqs,
+            queue_url=queue_url_alarm,
+            expected_topic_arn=topic_arn_alarm,
+            alarm_name=composite_alarm_name,
+            alarm_description=composite_alarm_description,
+            expected_state="ALARM",
+            expected_triggering_child_arn=alarm_2_arn,
+            expected_triggering_child_state="ALARM",
+        )
+
+        composite_alarms_list = aws_client.cloudwatch.describe_alarms(
+            AlarmNames=[composite_alarm_name], AlarmTypes=["CompositeAlarm"]
+        )
+        composite_alarm_in_alarm_when_alarm_2_in_alarm = composite_alarms_list["CompositeAlarms"][0]
+        snapshot.match(
+            "composite-alarm-in-alarm-when-alarm-2-is-in-alarm",
+            composite_alarm_in_alarm_when_alarm_2_in_alarm,
+        )
+
+        # trigger alarm 1 while alarm 2 is triggered - composite one shouldn't change
+        aws_client.cloudwatch.set_alarm_state(
+            AlarmName=alarm_1_name, StateValue="ALARM", StateReason="trigger alarm 1"
+        )
+
+        composite_alarms_list = aws_client.cloudwatch.describe_alarms(
+            AlarmNames=[composite_alarm_name], AlarmTypes=["CompositeAlarm"]
+        )
+        composite_alarm_not_changed_by_second_trigger = composite_alarms_list["CompositeAlarms"][0]
+        # It is not possible to reuse the snapshot within the same test,
+        # therefore checking that alarm reason hasn't changed instead of recording new snapshot
+        state_reason_data = json.loads(
+            composite_alarm_not_changed_by_second_trigger["StateReasonData"]
+        )
+        triggering_alarms = state_reason_data["triggeringAlarms"]
+        assert len(triggering_alarms) == 1
+        assert triggering_alarms[0]["arn"] == alarm_2_arn
```

Comment:
q: why do you think it's a problem to record new state?

Reply:
I thought that because composite alarm is unaffected it should be compared against previous snapshot, but now I can't remember why I thought it is a problem to record a new one. 🙂 🤷 I think I have a better understanding of snapshot testing by now and I changed this check to a new recorded snapshot comparison.